compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IrUtils.kt

```diff
@@ -515,11 +512,27 @@ val IrFunction.allTypeParameters: List<IrTypeParameter>
     else
         typeParameters
 
+
 fun IrMemberAccessExpression<*>.getTypeSubstitutionMap(irFunction: IrFunction): Map<IrTypeParameterSymbol, IrType> {
     val typeParameters = irFunction.allTypeParameters
-    return if (typeParameters.isEmpty()) emptyMap() else typeParameters.withIndex().associate {
+    val dispatchReceiverTypeArguments = (dispatchReceiver?.type as? IrSimpleType)?.arguments ?: emptyList()
+    if (typeParameters.isEmpty() && dispatchReceiverTypeArguments.isEmpty()) {
+        return emptyMap()
+    }
+
+    val result = mutableMapOf<IrTypeParameterSymbol, IrType>()
+    if (dispatchReceiverTypeArguments.isNotEmpty()) {
+        val parentTypeParameters = extractTypeParameters(irFunction.parentClassOrNull!!)
+        parentTypeParameters.withIndex().forEach { (index, typeParam) ->
+            dispatchReceiverTypeArguments[index].typeOrNull?.let {
+                result[typeParam.symbol] = it
+            }
+        }
+    }
+    irFunction.allTypeParameters.withIndex().associateTo(result) {
```

Comment:
(`associateTo` returns destination)
```suggestion
    return typeParameters.withIndex().associateTo(result) {
```

Reply:
Done, thanks!